File: test/py/case32_pathlib_extended.py
from pathlib import Path


def main() -> None:
    root = Path("test/obj/pathlib_case32")
    root.mkdir(parents=True, exist_ok=True)

    child = root / "values.txt"
    child.write_text("42")

    print(child.exists())
    print(child.name)
    print(child.stem)
    print((child.parent / "values.txt").exists())
    print(child.read_text())


if __name__ == "__main__":
    main()

File: test/cpp/case32_pathlib_extended.cpp
#include "cpp_module/gc.h"
#include "cpp_module/pathlib.h"
#include "cpp_module/py_runtime.h"
#include <algorithm>
#include <any>
#include <cstdint>
#include <fstream>
#include <ios>
#include <iostream>
#include <sstream>
#include <stdexcept>
#include <string>
#include <tuple>
#include <type_traits>
#include <unordered_map>
#include <unordered_set>
#include <vector>

using namespace std;
using namespace pycs::gc;

void py_main()
{
    Path root = Path("test/obj/pathlib_case32");
    root->mkdir(true, true);
    Path child = ((root) / ("values.txt"));
    child->write_text("42");
    py_print(child->exists());
    py_print(child->name());
    py_print(child->stem());
    py_print(((child->parent()) / ("values.txt"))->exists());
    py_print(child->read_text());
}

int main()
{
    py_main();
    return 0;
}
